### PR TITLE
[BUGFIX release] Fix container destroy timing

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

**Review by rwjblue — CHANGES_REQUESTED** [review]
Need to tweak some other parts of the container destroy lifecycle also here, I'll get those changes pushed up into this PR shortly...

File: packages/ember-runtime/lib/mixins/container_proxy.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ let containerProxyMixin = {
     this._super(...arguments);
 
     if (this.__container__) {
-      run(this.__container__, 'destroy');
+      schedule('destroy', this.__container__, 'destroy');
```
Comment (stefanpenner) [inline packages/ember-runtime/lib/mixins/container_proxy.js]: I think more `join` like behavior?

**Review by stefanpenner — DISMISSED** [review]

**Comment by rwjblue** [comment]
Just pushed an update that should address the concerns that @krisselden / @stefanpenner brought up.

Should be good to land once green...

**Review by wagenet — COMMENTED** [review]
Looks ok. Only point of confusion I had is that `finalizeDestroy` isn't automatically called. Is that as expected? Obviously, this is internal API, but I could see someone potentially calling only `destroy` and never calling `finalizeDestroy`.

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
> Only point of confusion I had is that finalizeDestroy isn't automatically called. Is that as expected? Obviously, this is internal API, but I could see someone potentially calling only destroy and never calling finalizeDestroy.

@wagenet - The container itself is not actually exposed, any user-land code calling `.destroy()` would be on the "owner" which is exactly the thing that mixes in `ContainerProxyMixin` and gets the `destroy` implementation in this PR.

**Pull request merged**